### ISSUE TITLE
Converting to Y-up coordinate system

### ISSUE THREAD

**Issue by braebo** [martinRenou/threejs-caustics#12]
TLDR:  What could I be missing in my attempts to rotate threejs-caustics from it's Z-up system into a ThreeJS scene using the default Y-up system?

I'm building an underwater scene and wanted to add caustics to the floor.  This incredible project seemed like the perfect solution!

I managed to convert this project to Typescript, separate everything into it's own files/folders, and add a series of controls to manupulate the uniforms / environment, and increase the size of the caustics:

<img width="600" alt="Screenshot 2022-07-04 at 7 13 59 PM" src="https://user-images.githubusercontent.com/55504972/177224494-ee106b20-73ad-4db4-a58e-90c589826273.png">

Problem is- when added to another ThreeJS scene, I noticed that everything was rotated by 90°.  It appears the scene is using a Z-up system instead of a Y-up system?

I've been converting as many things as I can find related to the Z axis to the Y axis, from the `lightProjection` camera and `WaterSimulation._camera` to references within each shader.  However, I can't seem to get anything other than this:

<img width="600" alt="Screenshot 2022-07-04 at 7 08 27 PM" src="https://user-images.githubusercontent.com/55504972/177224905-8e5d465a-0d02-4b4d-ab22-2f0183f614f2.png">

This is essentially the same result I get if I open `threejs-caustics` and add `scene.rotateX(-Math.PI / 2)`

I'm on day 3 of trying to figure this out so I figured it couldn't hurt to try to ask for a bit of advice 😅.  Either way, you've done incredible work on this project and I appreciate your generosity in sharing it with us!!

**Comment by FractalHQ** [comment]
OK so I think I finally figured it out!!!  These two lines needed to be changed from `lightPosition.xy` to `lightPosition.xz`:

https://github.com/martinRenou/threejs-caustics/blob/e24fa186406d0dccaa98e2055ebbede80f0350bb/shaders/environment/fragment.glsl#L36-L37

**Comment by braebo** [comment]
OK so I think I finally figured it out!!!  These two lines needed to be changed from `lightPosition.xy` to `lightPosition.xz`:

https://github.com/martinRenou/threejs-caustics/blob/e24fa186406d0dccaa98e2055ebbede80f0350bb/shaders/environment/fragment.glsl#L36-L37
